### ISSUE TITLE
To all experiencing clone errors

### ISSUE THREAD

**Issue by patrickmckenna** [githubteacher/feb-developers#208]
Many people are unable to use the https URL. This can be caused by many things, but one of the most common is that port 443 is blocked.

One alternative: trying using the SSH URL, after [setting up a key](https://help.github.com/articles/generating-an-ssh-key/) if needed.

**Comment by toddm111** [comment]
I got the following error when I tried to clone:
C:\view\git_training>git clone https://github.com/githubteacher/feb-developers.git
Cloning into 'feb-developers'...
fatal: unable to access 'https://github.com/githubteacher/feb-developers.git/': error:140770FC:SSL routines:SSL23_GET_SERVER_HELLO:unknown proto
col

**Comment by chandrapreeti** [comment]
using git version 2.5.3, when tried to clone, got error:
Cloning into 'feb-developers'...
fatal: unable to access 'https://github.com/githubteacher/feb-developers.git/':
error:140770FC:SSL routines:SSL23_GET_SERVER_HELLO:unknown protocol

**Comment by patrickmckenna** [comment]
@chandrapreeti that's the error you got after typing:

`git clone https://github.com/githubteacher/feb-developers.git/`?

**Comment by patrickmckenna** [comment]
@toddm111 Have you had a chance to try using SSH?

**Comment by toddm111** [comment]
I will try that at the next break.

**Comment by chandrapreeti** [comment]
Yes I could now clone it but lost significant part of trng for shell ..
Thanks!

Thanks,
Preeti

**Comment by jagmag** [comment]
Using windows, my "git push" did not work:

C:\tempgit\feb-developers [jagmag-hometown]> git push
remote: Permission to githubteacher/feb-developers.git denied to JG121180.
fatal: unable to access 'https://github.com/githubteacher/feb-developers.git/':
The requested URL returned error: 403
C:\tempgit\feb-developers [jagmag-hometown]>

**Comment by pkhayes** [comment]
C:\UsersDocuments\GitHub> git clone https://github.com/githubteacher/feb-developers.git
Cloning into 'feb-developers'...
fatal: unable to access 'https://github.com/githubteacher/feb-developers.git/': SSL certificate problem: self signed certificate

C:\Users\haypatri\Documents\GitHub> git clone https://github.com/githubteacher/feb-developers.git
Cloning into 'feb-developers'...
fatal: unable to access 'https://github.com/githubteacher/feb-developers.git/': Couldn't resolve proxy 'web-proxy'

**Comment by patrickmckenna** [comment]
@jagmag are you on a corporate network? Have you tried using the SSH URL? You can read more about your options [here](https://help.github.com/articles/which-remote-url-should-i-use/).

**Comment by patrickmckenna** [comment]
@pkhayes it sounds like you and others may need to chat with the IT department (assuming you're on your company's network), since this is probably caused by your network config.

**Comment by pkhayes** [comment]
C:\Users\haypatri\Documents\GitHub> git checkout -b gitphayes-more
fatal: Not a git repository (or any of the parent directories): .git

**Comment by patrickmckenna** [comment]
@pkhayes you need to be in the repo's directory itself before doing that, i.e. `cd C:\Users\haypatri\Documents\GitHub\feb-developers` first.

**Comment by pkhayes** [comment]
I could not download the feb-developers

**Comment by pkhayes** [comment]
C:\Users\haypatri\Documents\GitHub> cd C:\Users\haypatri\Documents\GitHub\
C:\Users\haypatri\Documents\GitHub> dir
C:\Users\haypatri\Documents\GitHub> git checkout master
fatal: Not a git repository (or any of the parent directories): .git
C:\Users\haypatri\Documents\GitHub>

**Comment by patrickmckenna** [comment]
> I could not download the feb-developers

@pkhayes you were unable to run the command

`git clone https://github.com/githubteacher/feb-developers`?

**Comment by pkhayes** [comment]
correct

**Comment by patrickmckenna** [comment]
@pkhayes ok, so that's a pre-requisite for doing anything else here. Were you able to try using SSH, or chat with IT?

**Comment by pkhayes** [comment]
Need to chit with IT.

**Comment by chandrapreeti** [comment]
Set proxy and resolved it

Thanks,
Preeti

**Comment by patrickmckenna** [comment]
Great to hear, @chandrapreeti!

**Comment by johnh-teradata** [comment]
I'm trying to use ssh now since the https isn't working.  I followed the instructions to create the key and it tests OK.  I'm still not able to do the clone.  Perhaps I'm doing something else wrong.  Any suggestions?

johnh@CS3096-01:~> ssh -T git@github.com
The authenticity of host 'github.com (192.30.252.129)' can't be established.
RSA key fingerprint is 16:27:ac:a5:76:28:2d:36:63:1b:56:4d:eb:df:a6:48 [MD5].
Are you sure you want to continue connecting (yes/no)? yes
Warning: Permanently added 'github.com,192.30.252.129' (RSA) to the list of known hosts.
**Hi johnh-teradata! You've successfully authenticated, but GitHub does not provide shell access.**
johnh@CS3096-01:~> cd git*
johnh@CS3096-01:~/github_training>
johnh@CS3096-01:~/github_training> git remote -v
fatal: Not a git repository (or any of the parent directories): .git
johnh@CS3096-01:~/github_training> git clone ssh://github.com/githubteacher/feb-developers.git
Cloning into 'feb-developers'...
Warning: Permanently added the RSA host key for IP address '192.30.252.128' to the list of known hosts.
Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
johnh@CS3096-01:~/github_training>

**Comment by patrickmckenna** [comment]
> johnh@CS3096-01:~/github_training> git clone ssh://github.com/githubteacher/feb-developers.git

@johnh-teradata try this for the URL:

`git@github.com:githubteacher/feb-developers.git`

**Comment by johnh-teradata** [comment]
Whoopee!  That worked!  Is that the approach I should use going forward, or is this just a debug step?  Thank you for your patience.

johnh@CS3096-01:~/github_training> git clone git@github.com:githubteacher/feb-developers.git
Cloning into 'feb-developers'...
Warning: Permanently added the RSA host key for IP address '192.30.252.130' to the list of known hosts.
remote: Counting objects: 1148, done.
remote: Compressing objects: 100% (850/850), done.
remote: Total 1148 (delta 332), reused 945 (delta 135), pack-reused 0
Receiving objects: 100% (1148/1148), 122.00 KiB | 0 bytes/s, done.
Resolving deltas: 100% (332/332), done.
Checking connectivity... done.
johnh@CS3096-01:~/github_training>

**Comment by patrickmckenna** [comment]
@johnh-teradata happy to help! Yes, I'd stick to using SSH for now. Just make sure to grab the appropriate URL on the repo homepage:

![image](https://cloud.githubusercontent.com/assets/3589536/13265256/c9038f7a-da26-11e5-9e7f-a8d5c0c47738.png)
